src/transformers/models/swiftformer/modeling_swiftformer.py

```diff
@@ -0,0 +1,638 @@
+        super().__init__()
+
+        self.to_query = nn.Linear(in_dims, token_dim * num_heads)
+        self.to_key = nn.Linear(in_dims, token_dim * num_heads)
+
+        self.w_g = nn.Parameter(torch.randn(token_dim * num_heads, 1))
+        self.scale_factor = token_dim**-0.5
+        self.proj = nn.Linear(token_dim * num_heads, token_dim * num_heads)
+        self.final = nn.Linear(token_dim * num_heads, token_dim)
+
+    def forward(self, x):
+        query = self.to_query(x)
+        key = self.to_key(x)
+
+        query = torch.nn.functional.normalize(query, dim=-1)
+        key = torch.nn.functional.normalize(key, dim=-1)
+
+        query_weight = query @ self.w_g
+        scaled_query_weight = query_weight * self.scale_factor
+        scaled_query_weight = scaled_query_weight.softmax(dim=-1)
+
+        global_queries = torch.sum(scaled_query_weight * query, dim=1)
+        global_queries = global_queries.unsqueeze(1).repeat(1, key.shape[1], 1)
+
+        out = self.proj(global_queries * key) + query
+        out = self.final(out)
+
+        return out
+
+
+class SwiftFormerLocalRepresentation(nn.Module):
+    """
+    Local Representation module for SwiftFormer that is implemented by 3*3 depth-wise and point-wise convolutions.
+    Input: tensor in shape [batch_size, channels, height, width] Output: tensor in shape [batch_size, channels, height,
+    width]
+    """
+
+    def __init__(self, config: SwiftFormerConfig, dim, kernel_size=3, drop_path=0.0, use_layer_scale=True):
+        super().__init__()
+        self.depth_wise_conv = nn.Conv2d(dim, dim, kernel_size=kernel_size, padding=kernel_size // 2, groups=dim)
+        self.norm = nn.BatchNorm2d(dim, eps=config.batch_norm_eps)
+        self.point_wise_conv1 = nn.Conv2d(dim, dim, kernel_size=1)
+        self.act = nn.GELU()
+        self.point_wise_conv2 = nn.Conv2d(dim, dim, kernel_size=1)
+        self.drop_path = SwiftFormerDropPath(drop_path) if drop_path > 0.0 else nn.Identity()
+        self.use_layer_scale = use_layer_scale
+        if use_layer_scale:
+            self.layer_scale = nn.Parameter(torch.ones(dim).unsqueeze(-1).unsqueeze(-1), requires_grad=True)
+
+    def forward(self, x):
+        input = x
+        x = self.depth_wise_conv(x)
+        x = self.norm(x)
+        x = self.point_wise_conv1(x)
+        x = self.act(x)
+        x = self.point_wise_conv2(x)
+        if self.use_layer_scale:
+            x = input + self.drop_path(self.layer_scale * x)
+        else:
+            x = input + self.drop_path(x)
+        return x
+
+
+class SwiftFormerEncoderBlk(nn.Module):
+    """
+    SwiftFormer Encoder Block for SwiftFormer. It consists of (1) Local representation module, (2)
+    SwiftFormerEfficientAdditiveAttention, and (3) MLP block. Input: tensor in shape [B, C, H, W] Output: tensor in
+    shape [B, C, H, W]
+    """
+
+    def __init__(
+        self,
+        config: SwiftFormerConfig,
+        dim,
+        drop=0.0,
+        drop_path=0.0,
+    ):
```

Comment:
* Can be put on one line
* `drop` not used 
* add types 

```suggestion
    def __init__(self, config: SwiftFormerConfig, dim: int,  drop_path: float = 0.0) -> None:
```